openapi/ogcapi-processes.bundled.json

```diff
@@ -159,6 +207,84 @@
             "$ref": "#/components/responses/NotFound"
           }
         }
+      },
+      "put": {
+        "summary": "replace a process.",
+        "description": "Replaces a process.\n\nFor more information, see [Section 6.4](http://docs.ogc.org/DRAFTS/20-044.html#_18582f42-ebc6-4284-9333-c089068f62b6).\n",
+        "operationId": "replace",
+        "tags": [
+          "DRU"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/components/parameters/processId-path"
+          }
+        ],
+        "requestBody": {
+          "description": "An OGC Application Package used to deploy a new process.",
+          "required": true,
+          "content": {
+            "application/ogcapppkg+json": {
+              "schema": {
+                "$ref": "#/components/schemas/ogcapppkg"
+              }
+            },
+            "application/cwl": {
+              "schema": {
+                "type": "string"
+              }
+            },
+            "application/cwl+json": {
+              "schema": {
+                "example": {}
+              }
+            }
```

Comment:
Same as other comment.

Reply:
@fmigneault I generated the bundle with:

```
swagger-cli bundle ogcapi-processes.yaml -o ogcapi-processes.bundled.json
```

from the very latest _master_ branch.

It should have included everything that has been included in `ogcapi-processes.yaml`.

I see `$ref: 'paths/processes-dru/pProcessListDeploy.yaml'` in there. Is there something else missing?

Reply:
Anything pointing at this commit https://github.com/opengeospatial/ogcapi-processes/commit/b47410e1375990f7ca76cac78f8aba2c8dd66f72 seems to be missing the corresponding changes of https://github.com/opengeospatial/ogcapi-processes/commit/f0743bbcefaea9b1206cb023db7f674219e80ee9

Reply:
@fmigneault I'm a fan of rebasing and trunk merging workflows.

It might be that something was lost in this merge commit 1297e3b3faf25fbb34855f2f835e98f9b662e0e1.

Could you please submit a new PR that fixes what was lost? Thanks for reviewing this.

cc. @gfenoy @pvretano

Reply:
It seems the is also a lot of duplicated contents in https://github.com/opengeospatial/ogcapi-processes/tree/master/openapi/paths/processes-dru
Some definitions are replicated in others. 
Why ones should be preferred? Individual HTTP method/path or the combined methods per corresponding paths?

Reply:
https://github.com/jerstlouis/ogcapi-processes/pull/1

Reply:
@fmigneault Regarding combined vs. individual, I believe it is because the OpenAPI includes does not allow to include at the individual method level, therefore we need different versions of the same path to allow implementations supporting Part 2 operations or not.

Reply:
@jerstlouis I guess you did not follow the instructions provided in the README.md for updating the following files:
 * pProcessListDeploy.yaml
 * pProcessDescriptionReplaceUndeploy.yaml

They should be updated before running the command to build the final bundle.

If it is of interest, I would be happy to prepare a PR including these modifications to have an up-to-date bundle.

Reply:
@gfenoy would it be possible to set things up so that we can simply run `swagger-cli bundle`?

Reply:
The commands for generating the proper version of the files should be run for every single modification to the following files:

* for generating: paths/processes-dru/pProcessListDeploy.yaml
  * paths/processes-core/pProcessList.yaml
  * paths/processes-dru/pDeploy.yaml

* for generating: paths/processes-dru/pProcessDescriptionReplaceUndeploy.yaml
  * paths/processes-core/pProcessDescription.yaml
  * paths/processes-dru/pReplace.yaml
  * paths/processes-dru/pUndeploy.yaml

Note, that once the pProcessListDeploy.yaml and pProcessDescriptionReplaceUndeploy.yaml are generated and the files listed here are not modified, then there is no reason for generating the file again.

So, to automate generating these two files would require to detect modifications in one of the listed files, run the required command and then commit these modifications. It can be run from a GitHub Action.

Let me know how should we proceed.

Maybe starting with a simple PR including the two correct files and then start a discussion on automating the updates can be done as an issue.

Reply:
@fmigneault @gfenoy I moved the commands to update.sh to make it easier to update. Hopefully this is all merged and good now. Thanks!